openssl/src/stack.rs

```diff
@@ -0,0 +1,275 @@
+use std::marker;
+use std::ops;
+use std::mem;
+use std::iter;
+
+use libc::c_void;
+
+use ffi;
+use opaque::Opaque;
+
+/// Trait implemented by stackable types. This must *only* be
+/// implemented on opaque types that can be directly casted into their
+/// `CType`.
+pub trait Stackable {
```

Comment:
Well there's one caveat: the `Stackable` trait contains the CType *and* the stack type (`st_stack_`...) this way I can have Rust check that it gets passed a correctly typed pointer in the constructor.

We could do without of course but we'd lose a bit of type-safety. And after all not all `OpenSslType`s are stackable, so maybe it would make sense to have a specific trait for the subset that is? Something like:

```rust
pub trait Stackable: OpenSslType {
    type StackType;
}
```